components/proxy/src/main/java/com/hotels/styx/infrastructure/FileBackedRegistry.java

```diff
@@ -103,7 +103,11 @@ public String fileName() {
                         return reloaded(format("%s, File reloaded.", logPrefix));
                     }
                 } catch (Exception e) {
-                    LOG.error("Not reloading {} as there was an error reading content", configurationFile.absolutePath(), e);
+                    // Eliminate unhelpful error message when resource constraint exception fails.
+                    // This is a candidate for further refactoring.
+                    if (!"Resource constraint failure".equals(e.getMessage())) {
```

Comment:
Relying on this message not to change introduces an undesirable coupling. I think the ideal solution is to change the `AbstractRegistry` to throw a specific class, which we can then check for.

Reply:
I specifically wanted to avoid adding a new exception class as it would have to be a new public class inside the Styx API module. This could potentially open up backwards compatibility issues.

The error handling for the styx registry/service startup classes, not just this PR, ought to be redesigned anyway. In my opinion we ought to look into introducing a monadic error handling into these classes, but that is another story.

My PR is supposed top be a quick work-around for a particular problem. I don't have time at the moment to refactor this much further. Therefore I left a comment in the source code. But we should definitely discuss this going forward.

Reply:
Mikko and myself talked about this solution being quite hacky but agreed it could be ok for the time being.  Would you find using a constant in both places  instead of a String literal acceptable enough?  

Another solution would be  creating an Exception for this case that is a subclass of IllegalStateException so we don't have compatibility issues, but you might argue it's a bit of a misuse the inheritance system.

Reply:
Yes I would agree with using a constant.